src/api/auth.py

```diff
@@ -1,20 +1,29 @@
-from fastapi import Security, HTTPException, status, Request
-from fastapi.security.api_key import APIKeyHeader
+from fastapi import Depends, HTTPException
+from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
+from supabase import create_client, Client
 import os
 import dotenv
 
 dotenv.load_dotenv()
 
-api_keys = []
+url: str = os.environ.get("SUPABASE_URL")
```

Comment:
```suggestion
url: str = os.environ.get("SUPABASE_API_URL")
```
going to nitpick a little at the new environment variable names since I found them a little confusing at first. I think generally the names should be close to how Supabase refers to it so that it's clear exactly what a database admin would need to find. URL on its own sounds vague. maybe this one would be more clear as `SUPABASE_API_URL`, since it's a Supabase project's URL for the REST API. Also, woah. didn't know direct code suggestions were a thing